diff --git a/slick.dataview.js b/slick.dataview.js
index 61a0f5e..43f580a 100644
--- a/slick.dataview.js
+++ b/slick.dataview.js
@@ -53,7 +53,11 @@
     var groupingInfoDefaults = {
       getter: null,
       formatter: null,
-      comparer: function(a, b) { return a.value - b.value; },
+      comparer: function(a, b) {
+        return (a.value === b.value ? 0 :
+          (a.value > b.value ? 1 : -1)
+        );
+      },
       predefinedValues: [],
       aggregators: [],
       aggregateEmpty: false,
