diff --git a/slick.groupitemmetadataprovider.js b/slick.groupitemmetadataprovider.js
index 18ee1c6..de05c72 100644
--- a/slick.groupitemmetadataprovider.js
+++ b/slick.groupitemmetadataprovider.js
@@ -82,7 +82,7 @@
         var range = _grid.getRenderedRange();
         this.getData().setRefreshHints({
           ignoreDiffsBefore: range.top,
-          ignoreDiffsAfter: range.bottom
+          ignoreDiffsAfter: range.bottom + 1
         });
 
         if (item.collapsed) {
@@ -106,7 +106,7 @@
             var range = _grid.getRenderedRange();
             this.getData().setRefreshHints({
               ignoreDiffsBefore: range.top,
-              ignoreDiffsAfter: range.bottom
+              ignoreDiffsAfter: range.bottom + 1
             });
 
             if (item.collapsed) {
